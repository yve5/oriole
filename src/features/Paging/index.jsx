const Paging = () => (
  <div className="card mt-3">
    <div className="card-header">Paging</div>

    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">First</th>
              <th scope="col">Last</th>
              <th scope="col">Handle</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">1</th>
              <td>Mark</td>
              <td>Otto</td>
              <td>@mdo</td>
            </tr>
            <tr>
              <th scope="row">2</th>
              <td>Jacob</td>
              <td>Thornton</td>
              <td>@fat</td>
            </tr>
            <tr>
              <th scope="row">3</th>
              <td colSpan="2">Larry the Bird</td>
              <td>@twitter</td>
            </tr>
          </tbody>
        </table>

        <nav aria-label="Page navigation">
          <ul className="pagination">
            {['Previous', '1', '2', '3', 'Next'].map((item) => (
              <li className="page-item" key={item}>
                <button className="page-link" type="button">
                  {item}
                </button>
              </li>
            ))}
          </ul>
        </nav>
      </li>
    </ul>
  </div>
);

export default Paging;
