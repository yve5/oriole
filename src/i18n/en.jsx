const I18N_DICTIONARY = {
  LANG: 'en',
  LANGUAGE: 'English',
  DATE_FORMAT: 'MM/dd/yyyy',
  LOCALE: 'enUS',
  'About us': 'About us',
  'An error has been detected. Please try again.':
    'An error has been detected. Please try again.',
  Buttons: 'Buttons',
  Changelog: 'Changelog',
  Collapse: 'Collapse',
  Expand: 'Expand',
  Internationalization: 'Internationalization',
  Paging: 'Paging',
  'The request has been sent successfully.':
    'The request has been sent successfully.',
};

export default I18N_DICTIONARY;
