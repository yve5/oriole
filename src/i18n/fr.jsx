const I18N_DICTIONARY = {
  LANG: 'fr',
  LANGUAGE: 'Français',
  DATE_FORMAT: 'dd/MM/yyyy',
  LOCALE: 'fr',
  'About us': 'À propos de nous',
  'An error has been detected. Please try again.':
    'Une erreur a été détectée. Merci de recommencer.',
  Buttons: 'Boutons',
  Changelog: 'Journal des modifications',
  'Changes have been saved.': 'Les changements ont été sauvegardés.',
  'Choose a file': 'Choisir un fichier',
  Collapse: 'Réduire',
  Columns: 'Colonnes',
  'Creation date': 'Date de création',
  'Descending order': 'Ordre décroissant',
  Disabled: 'Désactivé',
  Enabled: 'Activé',
  Expand: 'Développer',
  First: 'Premier',
  'Hello World': 'Bonjour le monde',
  'Hide columns': 'Cacher les colonnes',
  Internationalization: 'Internationalisation',
  Paging: 'Pagination',
  'The request has been sent successfully.':
    'La demande a été envoyée avec succès.',
};

export default I18N_DICTIONARY;
