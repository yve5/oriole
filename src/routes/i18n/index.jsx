import I18nPage from './components/I18n';

export default I18nPage;
