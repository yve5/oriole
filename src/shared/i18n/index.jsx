import LangSwitch from './components/LangSwitch';
import I18n from './components/I18n';
import i18nReducer from './reducer';

export { I18n, LangSwitch, i18nReducer };
