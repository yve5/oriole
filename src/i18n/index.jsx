import fr from './fr';
import en from './en';
import { tsl } from '../shared/i18n/utils/Translate';

const i18n = {
  tsl: (term, args) => tsl(fr, term, args),
  dictionaries: [fr, en],
  lang: fr.LANG,
};

export default i18n;
