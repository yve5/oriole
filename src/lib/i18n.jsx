import { I18n, LangSwitch, i18nReducer } from '../shared/i18n';

export { I18n, LangSwitch, i18nReducer };
