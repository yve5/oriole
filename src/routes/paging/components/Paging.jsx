import React from 'react';
import { I18n } from '../../../shared/i18n';

const Paging = () => (
  <div className="card">
    <h1>
      <I18n>Hello World</I18n>
    </h1>

    <ul className="list-group list-group-flush">
      <li className="list-group-item">
        <table className="table">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">First</th>
              <th scope="col">Last</th>
              <th scope="col">Handle</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">1</th>
              <td>Mark</td>
              <td>Otto</td>
              <td>@mdo</td>
            </tr>
            <tr>
              <th scope="row">2</th>
              <td>Jacob</td>
              <td>Thornton</td>
              <td>@fat</td>
            </tr>
            <tr>
              <th scope="row">3</th>
              <td colSpan="2">Larry the Bird</td>
              <td>@twitter</td>
            </tr>
          </tbody>
        </table>

        <nav>
          <ul className="pagination justify-content-center">
            <li className="page-item">
              <button className="page-link" type="button">
                <span aria-hidden="true">&laquo;</span>
              </button>
            </li>
            {[1, 2, 3].map((item) => (
              <li className="page-item" key={item}>
                <button className="page-link" type="button">
                  {item}
                </button>
              </li>
            ))}
            <li className="page-item">
              <button className="page-link" type="button">
                <span aria-hidden="true">&raquo;</span>
              </button>
            </li>
          </ul>
        </nav>
      </li>
    </ul>
  </div>
);

export default Paging;
