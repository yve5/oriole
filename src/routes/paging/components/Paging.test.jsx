import React from 'react';
import configureMockStore from 'redux-mock-store';

import { thunk } from 'redux-thunk';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import Paging from './Paging';
import i18n from '../../../i18n';

const mockStore = configureMockStore([thunk]);

describe('Paging', () => {
  it('should match expected snapshot', async () => {
    const instance = (
      <Provider store={mockStore({ i18n })}>
        <Paging />
      </Provider>
    );

    const { asFragment, rerender } = render(instance);

    rerender(instance);

    expect(asFragment()).toMatchSnapshot();
  });
});
