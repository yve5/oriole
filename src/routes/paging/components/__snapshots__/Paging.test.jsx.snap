// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Paging > should match expected snapshot 1`] = `
<DocumentFragment>
  <h1
    class="title title--alone"
  >
    Pagination
  </h1>
  <div
    class="card"
  >
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
        <table
          class="table"
        >
          <thead>
            <tr>
              <th
                scope="col"
              >
                #
              </th>
              <th
                scope="col"
              >
                First
              </th>
              <th
                scope="col"
              >
                Last
              </th>
              <th
                scope="col"
              >
                Handle
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th
                scope="row"
              >
                1
              </th>
              <td>
                Mark
              </td>
              <td>
                Otto
              </td>
              <td>
                @mdo
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                2
              </th>
              <td>
                Jacob
              </td>
              <td>
                Thornton
              </td>
              <td>
                @fat
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                3
              </th>
              <td
                colspan="2"
              >
                Larry the Bird
              </td>
              <td>
                @twitter
              </td>
            </tr>
          </tbody>
        </table>
        <nav>
          <ul
            class="pagination justify-content-center"
          >
            <li
              class="page-item"
            >
              <button
                class="page-link"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  «
                </span>
              </button>
            </li>
            <li
              class="page-item"
            >
              <button
                class="page-link"
                type="button"
              >
                1
              </button>
            </li>
            <li
              class="page-item"
            >
              <button
                class="page-link"
                type="button"
              >
                2
              </button>
            </li>
            <li
              class="page-item"
            >
              <button
                class="page-link"
                type="button"
              >
                3
              </button>
            </li>
            <li
              class="page-item"
            >
              <button
                class="page-link"
                type="button"
              >
                <span
                  aria-hidden="true"
                >
                  »
                </span>
              </button>
            </li>
          </ul>
        </nav>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
